modules/zivid/experimental/projection.py

```diff
@@ -0,0 +1,147 @@
+
+        Args:
+            impl: Reference to internal/back-end instance.
+
+        Raises:
+            TypeError: If argument does not match the expected internal class.
+        """
+        if isinstance(impl, _zivid.ProjectedImage):
+            self.__impl = impl
+        else:
+            raise TypeError(
+                "Unsupported type for argument impl. Got {}, expected {}.".format(
+                    type(impl), type(_zivid.ProjectedImage)
+                )
+            )
+
+    def __str__(self):
+        return str(self.__impl)
+
+    def capture(self, settings2d):
+        """Capture a single 2D frame without stopping the ongoing image projection.
+
+        This function can only be used with a zero-brightness 2D capture, otherwise it
+        will interfere with the projected image. An exception will be thrown if settings
+        contains brightness > 0 or if brightness is not set at all.
+
+        Args:
+            settings2d: A Settings2D instance to be used for 2D capture.
+
+        Returns:
+            A Frame2D containing a 2D image plus metadata.
+
+        Raises:
+            TypeError: If argument is not a Settings2D.
+        """
+
+        if isinstance(settings2d, Settings2D):
+            return Frame2D(self.__impl.capture(_to_internal_settings2d(settings2d)))
+        raise TypeError("Unsupported settings type: {}".format(type(settings2d)))
+
+    def stop(self):
+        """Stop the ongoing image projection."""
+        self.__impl.stop()
+
+    def active(self):
+        """Check if a handle is associated with an ongoing image projection.
+
+        Returns:
+            A boolean indicating projection state.
+        """
+        return self.__impl.active()
+
+    def release(self):
+        """Release the underlying resources and stop projection."""
+        try:
+            impl = self.__impl
+        except AttributeError:
+            pass
+        else:
+            impl.release()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.release()
+
+    def __del__(self):
+        self.release()
+
+
+def projector_resolution(camera):
+    """Get the resolution of the internal projector in the Zivid camera.
+
+    Args:
+        camera: The Camera instance to get the projector resolution of.
+
+    Returns:
+        The resolution as a tuple (height,width).
+    """
+    return _zivid.projection.projector_resolution(
+        camera._Camera__impl  # pylint: disable=protected-access
+    )
+
+
+def show_image(camera, image_bgra):
+    """Display a 2D color image using the projector.
+
+    This function returns a ProjectedImage instance. Projection will continue until this object is
+    destroyed or if its stop() method is called. Alternatively it can be used as a context manager,
+    in which case the projection will stop when exiting the "with" block.
```

Comment:
The c++ docstring mentions that this projection will also be cancelled if another capture is instantiated on the camera, but that is not mentioned here.

Reply:
Did my best. Can't be identical, since Python is different from C++.